Add k8s 1.19 provider

dhiller opened the pull request:
Create k8s 1.19 folder and files for provisioning.

Use iptables instead of ipvs for now.

Update sonobuoy config and version to 0.19.0, use different version
per provider as given per config file conformance.json.

Also add SONOBUOY_EXTRA_ARGS to be able to execute certain tests.

See https://sonobuoy.io/docs/master/sonobuoy-config/ for details

dhiller commented:
/test check-provision-k8s-1.19

dhiller commented on hack/conformance.sh:
```diff
@@ -3,7 +3,7 @@
 
 ARTIFACTS=${ARTIFACTS:-${PWD}}
-sonobuoy_version=0.18.2
+sonobuoy_version=0.19.0
```
Good point. Uses a version file now.

dhiller commented:
/test check-provision-k8s-1.19
/test check-provision-k8s-1.18

dhiller commented:
/test check-provision-k8s-1.19
/test check-provision-k8s-1.18

dhiller commented:
/test check-provision-k8s-1.19
/test check-provision-k8s-1.18

dhiller commented:
/test check-provision-k8s-1.19
/test check-provision-k8s-1.18

dhiller commented:
/test check-provision-k8s-1.19
/test check-provision-k8s-1.18

xpivarc commented:
/lgtm 
This should allow us testing new lane and we can investigate the IPVS problem :) :+1:

dhiller commented:
Created kubevirt/kubevirtci#439 to investigate in ipvs test case failure.

dhiller commented:
/test push-k8s-1.19

xpivarc commented:
/lgtm
/cc @danielBelenky

oshoval commented:
can you add to the description the reason of "Furthermore set nodes to 3" please?

oshoval commented:
Nice,
can you add to the description the reason of "Furthermore set nodes to 3" please?

oshoval commented on cluster-provision/k8s/check-cluster-up.sh:
```diff
@@ -18,7 +21,7 @@ function cleanup {
   export KUBEVIRTCI_PROVISION_CHECK=1
   export KUBEVIRT_PROVIDER="k8s-${provision_dir}"
-  export KUBEVIRT_NUM_NODES=2
+  export KUBEVIRT_NUM_NODES=3
```
it affect all versions from now on, in this file, we want it also on < 19 ?

dhiller replied:
I think we can leave it like that, this is only for testing.

oshoval replied:
up to you, i just wonder how much more time it takes,
can you please update the description with the reason ?

qinqon replied:
Agree, this is not relevant, let's remove it.

dhiller commented:
> Also i would suggest please to create a commit with the file copying, and another one with the custom changes if possible ?

Please note that upon merge the commits will all get squashed together: https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml#L72

oshoval commented:
> > Also i would suggest please to create a commit with the file copying, and another one with the custom changes if possible ?
> 
> Please note that upon merge the commits will all get squashed together: https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/config.yaml#L72

You right, but at least in the PR it would still be visible ?

oshoval commented on cluster-provision/k8s/1.19/provision.sh:
```diff
@@ -0,0 +1,318 @@
+for i in $(grep -A 2 "IMAGE" /provision/local-volume.yaml | grep value | awk -F\" '{print $2}'); do docker_pull_retry $i; done
+
+# Create a properly labelled tmp directory for testing
+mkdir -p /provision/kubevirt.io/tests
```
see https://github.com/kubevirt/kubevirtci/pull/436 please
those line should be adapted as well

dhiller replied:
Good point, will update.

qinqon commented on cluster-provision/k8s/1.16/sonobuoy_version:
```diff
@@ -0,0 +1 @@
+0.18.2
```
conformance version is inside `conformance.json` we don't need new file, `jq .Version` will give it to us.

dhiller replied:
Not sure whether we have jq installed in test image. Let me check.

oshoval replied:
we usually use a containerized version of it if needed, Quique added it once

qinqon commented on cluster-provision/k8s/1.17/sonobuoy_version:
```diff
@@ -0,0 +1 @@
+0.18.2
```
ditto

qinqon commented on cluster-provision/k8s/1.18/sonobuoy_version:
```diff
@@ -0,0 +1 @@
+0.18.2
```
ditto

qinqon commented on cluster-provision/k8s/1.19/provision.sh:
```diff
@@ -0,0 +1,318 @@
+dnf update -y kernel
+
+# add for debugging
+dnf install -y ipvsadm
```
Not needed anymore since we are moving to iptables

qinqon commented on cluster-provision/k8s/1.19/sonobuoy_version:
```diff
@@ -0,0 +1 @@
+0.19.0
```
ditto

qinqon commented on cluster-provision/k8s/check-cluster-up.sh:
```diff
@@ -6,6 +6,9 @@ set -exuo pipefail
 CI=${CI:-"false"}
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 provision_dir="$1"
+SONOBUOY_VERSION=0.19.0
+[ -f "$DIR/$provision_dir/sonobuoy_version" ] && SONOBUOY_VERSION=$(cat "$DIR/$provision_dir/sonobuoy_version" | tr -d '\n')
+export SONOBUOY_VERSION
```
fail if there is no version and extract version from conformance.json with `jq .Version`

qinqon commented on hack/conformance.sh:
```diff
@@ -41,6 +41,11 @@ trap teardown EXIT
 
 run_cmd="./sonobuoy run --wait"
 
+SONOBUOY_FOCUS=${SONOBUOY_FOCUS:-}
+if [ -n "$SONOBUOY_FOCUS" ]; then
+    run_cmd+=" --e2e-focus ${SONOBUOY_FOCUS// /_}"
+fi
+
```
Let's do a SONOBUOY_EXTRA_ARGS so we can put whatever we want, it has plenty or arguments.

qinqon reviewed (changes requested):
Let's take sonobuoy version from conformance.json.

dhiller commented:
/test push-k8s-1.19

dhiller commented:
@oshoval @qinqon addressed your comments, PTAL. Only changes between 1.18 and 1.19 is iptables vs. ipvs, cluster version and conformance.json sonobuoy config now. Otherwise please see updated PR description.

dhiller commented:
/hold need to wait for the push and update the shasum after the push is ready

jean-edouard commented:
@dhiller shouldn't this effort include removing 1.16?

dhiller commented:
> @dhiller shouldn't this effort include removing 1.16?

Good question. Not sure as I don't know how much the user community for kubevirtci still requires the provider, which they will lose when they bump?

vladikr commented:
> > @dhiller shouldn't this effort include removing 1.16?
> 
> Good question. Not sure as I don't know how much the user community for kubevirtci still requires the provider, which they will lose when they bump?

I know that we agreed to support 3 k8s version back. I'd love to get rid of the 1.16 provider. It's based on centos7 (unlike other providers) and doesn't use the same virt-launcher version as other providers.
@davidvossel what do you think? Can we just remove 1.16 when 1.19 is ready?

xpivarc commented:
> @dhiller shouldn't this effort include removing 1.16?

We should but lets wait until we make 1.19 mandatory.

dhiller commented:
As we saw from a job run of PR kubevirt/kubevirt#4135 [here](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4135/pull-kubevirt-e2e-k8s-1.19/1303238153573765120) it looks like we have some problems within kubevirt when using the provider.

I think we should wait until we have those sorted out. Created issue kubevirt/kubevirtci#441 to track that.

oshoval commented:
btw would be great imo if we stop using squash on kubevirtci (not part of this PR of course, but it was raised here, and in the past)

davidvossel commented:
> what do you think? Can we just remove 1.16 when 1.19 is ready?

yes

dhiller commented:
/hold cancel

The k8s-1.19 provider passes all conformance tests now, it has the latest pushed sha for 1.19, so we should be ready to merge it now.

qinqon commented:
/lgtm
/approve
